### ISSUE TITLE
[HELP] Permission Denied 

### ISSUE THREAD

**Issue by pen-pal** [CrowdStrike/falcon-helm#110]
While running falcon helm with default setting it fails with `cannot create directory /opt/CrowdStrike: Permission deined`. The issue is observed on `node.container`
does it require any additional roles and permission in order to achieve that?
![image](https://user-images.githubusercontent.com/61139563/160663103-4425fd4b-281e-4eec-9868-01b4813a7d77.png)

**Comment by isimluk** [comment]
This is not what normally happens.

 - Is there anything particular worth nothing about the cluster?
 - Is this some kind of kubernetes as a service offering (like EKS Fargate)?
 - Is there some hardening applied to the cluster worth mentioning

**Comment by redhatrises** [comment]
Also, what sensor are you trying to deploy to the nodes? falcon-container or falcon-linux?

**Comment by dempti** [comment]
> This is not what normally happens.
> 
> * Is there anything particular worth nothing about the cluster?
> * Is this some kind of kubernetes as a service offering (like EKS Fargate)?
> * Is there some hardening applied to the cluster worth mentioning?

I tried to install it on EKS server and it's eks managed node groups.Everything is as per eks-managed-node-groups itself and nothing else has been altered.

**Comment by dempti** [comment]
> Also, what sensor are you trying to deploy to the nodes? falcon-container or falcon-linux?

I am trying to install falcon-container.

The sidecar container injection deployed, however it is also throwing error while trying to inject it on other pods. I did saw the port required is 443, but still quite not clear regarding the requirement of it though.
```
31m         Warning   FailedCreate   replicaset/nginx-6799fc88d8                     Error creating: Internal error occurred: failed calling webhook "falcon-sensor-injector.falcon-system.svc": Post "https://falcon-sensor-injector.falcon-system.svc:443/mutate?timeout=30s": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
14m         Warning   FailedCreate   replicaset/nginx-6799fc88d8                     Error creating: Internal error occurred: failed calling webhook "falcon-sensor-injector.falcon-system.svc": Post "https://falcon-sensor-injector.falcon-system.svc:443/mutate?timeout=30s": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

**Comment by pen-pal** [comment]
> This is not what normally happens.
> 
> * Is there anything particular worth nothing about the cluster?
> * Is this some kind of kubernetes as a service offering (like EKS Fargate)?
> * Is there some hardening applied to the cluster worth mentioning?

I tried to install it on EKS server and it's eks managed node groups.Everything is as per eks-managed-node-groups itself and nothing else has been altered.

To be more precise here is the exact EKS cluster setup config. Redacting few info related to vpc and other
```
{
    "cluster_name": "eks",
    "cluster_version": "1.21",
    "cluster_endpoint_private_access": true,
    "cluster_endpoint_public_access": true,
    "worker_create_security_group": true,
    "worker_create_cluster_primary_security_group_rules": true,
    "enable_irsa": true,
    "cluster_addons": {
      "coredns": {
        "resolve_conflicts": "OVERWRITE"
      },
      "vpc-cni": {
        "resolve_conflicts": "OVERWRITE"
      },
      "kube-proxy": {
        "resolve_conflicts": "OVERWRITE"
      }
    },
    "node_security_group_additional_rules": {
      "egress_http": {
        "description": "Worker nodes access internet through http",
        "protocol": "tcp",
        "from_port": 80,
        "to_port": 80,
        "type": "egress",
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "ipv6_cidr_blocks": [
          "::/0"
        ]
      },
      "egress_postgres": {
        "description": "Outbound for postgres instances",
        "protocol": "tcp",
        "from_port": 5432,
        "to_port": 5432,
        "type": "egress",
        "cidr_blocks": [
          "0.0.0.0/0"
        ],
        "ipv6_cidr_blocks": [
          "::/0"
        ]
      }
    },
    "eks_managed_node_groups_defaults": {
      "disk_size": 50,
      "tags": {
        "k8s.io/cluster-autoscaler/${local.cluster_name}": "owned",
        "k8s.io/cluster-autoscaler/enabled": "TRUE"
      }
    },
    "eks_managed_node_groups": {
      "eks_node_group": {
        "post_bootstrap_user_data": "cd /tmp\n sudo yum install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm\n sudo systemctl enable amazon-ssm-agent\n sudo systemctl start amazon-ssm-agent\n",
        "min_size": 1,
        "max_size": 3,
        "desired_size": 1,
        "force_update_version": true,
        "instance_types": [
          "t3.medium"
        ],
        "enable_monitoring": true,
        "ebs_optimized": true,
        "create_iam_role": true,
        "iam_role_name": "eks-managed-node-group-complete-example",
        "iam_role_use_name_prefix": false,
        "iam_role_description": "EKS managed node group complete example role",
        "iam_role_tags": {
          "Purpose": "Protector of the kubelet"
        },
        "iam_role_additional_policies": [
          "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly"
        ]
      }
    }
  }
```

**Comment by pen-pal** [comment]
> Also, what sensor are you trying to deploy to the nodes? falcon-container or falcon-linux?

I am trying to install falcon-container.

The sidecar container injection deployed, however it is also throwing error while trying to inject it on other pods. I did saw the port required is 443, but still quite not clear regarding the requirement of it though.
```
31m         Warning   FailedCreate   replicaset/nginx-6799fc88d8                     Error creating: Internal error occurred: failed calling webhook "falcon-sensor-injector.falcon-system.svc": Post "https://falcon-sensor-injector.falcon-system.svc:443/mutate?timeout=30s": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
14m         Warning   FailedCreate   replicaset/nginx-6799fc88d8                     Error creating: Internal error occurred: failed calling webhook "falcon-sensor-injector.falcon-system.svc": Post "https://falcon-sensor-injector.falcon-system.svc:443/mutate?timeout=30s": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

**Comment by redhatrises** [comment]
@pen-pal please make sure that port 4433 (unless you changed it to 443) isn't blocked to internal cluster configurations.

**Comment by pen-pal** [comment]
> @pen-pal please make sure that port 4433 (unless you changed it to 443) isn't blocked to internal cluster configurations.

sure.

What about the permission denied issue for `/opt/CrowdStrike` directory creation?

**Comment by redhatrises** [comment]
> > @pen-pal please make sure that port 4433 (unless you changed it to 443) isn't blocked to internal cluster configurations.
> 
> sure.
> 
> What about the permission denied issue for `/opt/CrowdStrike` directory creation?

This looks like the `falcon-container` sensor is being used on the nodes, am I correct? If so and you are wanting to protect the nodes while not using the sidecar, the `falcon-linux` container sensor should be used on the nodes instead of `falcon-container`. If you want to use the sidecar and not protect the nodes, make sure `node.enabled` is set to `false`.

**Comment by pen-pal** [comment]
> > > @pen-pal please make sure that port 4433 (unless you changed it to 443) isn't blocked to internal cluster configurations.
> > 
> > 
> > sure.
> > What about the permission denied issue for `/opt/CrowdStrike` directory creation?
> 
> This looks like the `falcon-container` sensor is being used on the nodes, am I correct? If so and you are wanting to protect the nodes while not using the sidecar, the `falcon-linux` container sensor should be used on the nodes instead of `falcon-container`. If you want to use the sidecar and not protect the nodes, make sure `node.enabled` is set to `false`.

On the sensor download, I can only see as shown in the image below. For, node container, where should I download it from ?

![image](https://user-images.githubusercontent.com/61139563/161367131-36122942-44fd-4024-b3db-550542acd3f9.png)

**Comment by redhatrises** [comment]
@pen-pal The sensor download page no longer contains the latest sensor downloads. Please use registry.crowdstrike.com to pull and get the latest sensor. The Falcon Container cannot be used for nodes which is why you are getting the error. Please see https://falcon.crowdstrike.com/documentation/20/falcon-sensor-for-linux#deploying-falcon-sensor-for-linux-as-a-daemon-set-on-a-kubernetes-cluster for node deployment which uses the falcon-linux sensor.

**Comment by pen-pal** [comment]
> @pen-pal The sensor download page no longer contains the latest sensor downloads. Please use registry.crowdstrike.com to pull and get the latest sensor. The Falcon Container cannot be used for nodes which is why you are getting the error. Please see https://falcon.crowdstrike.com/documentation/20/falcon-sensor-for-linux#deploying-falcon-sensor-for-linux-as-a-daemon-set-on-a-kubernetes-cluster for node deployment which uses the falcon-linux sensor.

@redhatrises thank you.
will try this out and get back in touch

**Comment by redhatrises** [comment]
@pen-pal were you successful?

**Comment by pen-pal** [comment]
@redhatrises sorry for the delay.
Not exactly, I followed the steps you mentioned but it seems to fail while pulling the image
While following the steps, the settings looks like
```
FALCON_IMAGE_REPO=registry.crowdstrike.com/falcon-sensor/eu-1/release/falcon-sensor
FALCON_IMAGE_SECRET=cs-registry-secret
FALCON_IMAGE_TAG=6.39.0-13601.falcon-linux.x86_64.Release.api.eu-1
```

But it fails on pulling this image from the registry.

**Comment by redhatrises** [comment]
@pen-pal please check the TAG, it doesn't look right. IIRC, it should be `6.39.0-13601.falcon-linux.x86_64.Release.EU-1`

**Comment by dempti** [comment]
@redhatrises thank you.
but seems the issue was with the credentials, its throwing up issue related to authentication.
Here is what I did similar to the docs,
1. copy customer id check sum and add it as  a ART_USERNAME="fc-cid"
2. use token genereated from `curl -i -X GET -H "authorization: Bearer ${TOKEN}" https://api.eu-1.crowdstrike.com/container-security/entities/image-registry-credentials/v1`

but while trying to execute docker login it fails.

Is there something related to the username? Do we use the client id generated from api keys as a username? I have followed the docs step by step without missing anything.
